server/types.py

```diff
@@ -1,5 +1,23 @@
-from typing import NamedTuple
+from typing import NamedTuple, Optional
 
-# Represents a peer IP address
-Address = NamedTuple('Address', [('host', str), ('port', int)])
-Address.from_string = lambda s: Address(*(s.split(':')))
+
+class Address(NamedTuple):
+    """A peer IP address"""
+
+    host: str
+    port: int
+
+    @classmethod
+    def from_string(cls, s: str) -> "Address":
+        host, port = s.split(":")
```

Comment:
please, don't name `s`, name it as `string_address`

Reply:
I would use `rsplit` with limitation on 2 peaces (checkou docs, imho it's something like `rsplit(":", 1)`